test/common/quic/envoy_quic_session_cache_test.cc

```diff
@@ -0,0 +1,343 @@
+    "010b050003820101008c1f1e380831b6437a8b9284d28d4ead38d9503a9fc936db89048aa2"
+    "edd6ec2fb830d962ef7a4f384e679504f4d5520f3272e0b9e702b110aff31711578fa5aeb1"
+    "11e9d184c994b0f97e7b17d1995f3f477f25bc1258398ec0ec729caed55d594a009f48093a"
+    "17f33a7f3bb6e420cc3499838398a421d93c7132efa8bee5ed2645cbc55179c400da006feb"
+    "761badd356cac3bd7a0e6b22a511106a355ec62a4c0ac2541d2996adb4a918c866d10c3e31"
+    "62039a91d4ce600b276740d833380b37f66866d261bf6efa8855e7ae6c7d12a8a864cd9a1f"
+    "4663e07714b0204e51bbc189a2d04c2a5043202379ff1c8cbf30cbb44fde4ee9a1c0c976dc"
+    "4943df2c132ca4020400aa7f047d494e534543555245003072020101020203040402130104"
+    "000420d87ed2ab8cafc986fd2e288bd2d654cd57c3a2bed1d53220726e55fed39d021ea106"
+    "02045ed16771a205020302a300a4020400b20302011db5060404bd909308b807020500ffff"
+    "ffffb9050203093a80ba07040568332d3238bb030101ffbc03040100b20302011db3820307"
+    "30820303308201eba003020102020102300d06092a864886f70d01010b050030243110300e"
+    "060355040a130741636d6520436f3110300e06035504031307526f6f74204341301e170d31"
+    "33303130313130303030305a170d3233313233313130303030305a302c3110300e06035504"
+    "0a130741636d6520436f311830160603550403130f496e7465726d65646961746520434130"
+    "820122300d06092a864886f70d01010105000382010f003082010a0282010100cd3550e70a"
+    "6880e52bf0012b93110c50f723e1d8d2ed489aea3b649f82fae4ad2396a8a19b31d1d64ab2"
+    "79f1c18003184154a5303a82bd57109cfd5d34fd19d3211bcb06e76640e1278998822dd72e"
+    "0d5c059a740d45de325e784e81b4c86097f08b2a8ce057f6b9db5a53641d27e09347d993ee"
+    "acf67be7d297b1a6853775ffaaf78fae924e300b5654fd32f99d3cd82e95f56417ff26d265"
+    "e2b1786c835d67a4d8ae896b6eb34b35a5b1033c209779ed0bf8de25a13a507040ae9e0475"
+    "a26a2f15845b08c3e0554e47dbbc7925b02e580dbcaaa6f2eecde6b8028c5b00b33d44d0a6"
+    "bfb3e72e9d4670de45d1bd79bdc0f2470b71286091c29873152db4b1f30203010001a33830"
+    "36300e0603551d0f0101ff04040302020430130603551d25040c300a06082b060105050703"
+    "01300f0603551d130101ff040530030101ff300d06092a864886f70d01010b050003820101"
+    "00bc4f8234860558dd404a626403819bfc759029d625a002143e75ebdb2898d1befdd326c3"
+    "4b14dc3507d732bb29af7e6af31552db53052a2be0d950efee5e0f699304231611ed8bf73a"
+    "6f216a904c6c2f1a2186d1ed08a8005a7914394d71e7d4b643c808f86365c5fecad8b52934"
+    "2d3b3f03447126d278d75b1dab3ed53f23e36e9b3d695f28727916e5ee56ce22d387c81f05"
+    "919b2a37bd4981eb67d9f57b7072285dbbb61f48b6b14768c069a092aad5a094cf295dafd2"
+    "3ca008f89a5f5ab37a56e5f68df45091c7cb85574677127087a2887ba3baa6d4fc436c6e40"
+    "40885e81621d38974f0c7f0d792418c5adebb10e92a165f8d79b169617ff575c0d4a85b506"
+    "0404bd909308b603010100b70402020403b807020500ffffffffb9050203093a80ba070405"
+    "68332d3238bb030101ff";
+
+class FakeTimeSource : public TimeSource {
+public:
+  // From TimeSource.
+  SystemTime systemTime() override { return fake_time_; }
+  MonotonicTime monotonicTime() override { return MonotonicTime(); }
```

Comment:
mind commenting why you can't just use simtime's monotonicTime too?

Reply:
Done